app/src/main/java/org/glucosio/android/activity/ExternalLinkActivity.java

```diff
@@ -89,13 +89,20 @@ public void setupToolbarTitle(String link) {
     }
 
     @Override
-    public String extractAction() {
-        Bundle p;
+    public String extractTitle() {
+        String title = "";
         if (getIntent().getExtras() != null) {
-            p = getIntent().getExtras();
-            return p.getString("key");
+            title = getIntent().getStringExtra(TITLE_KEY);
         }
-        return null;
+        return title;
+    }
+
+    @Override public String extractUrl() {
```

Comment:
Agree, I didn't notice it is part of contract between presenter and view. So in this case it is matter of taste:
 1. Keep contract and have one general private method in activity wich knows about paramters
 2. Make parameters defined in contract and use them everywhere

As for me the first one is cleaner and preferrable